subsys/cpp/Kconfig

```diff
@@ -5,6 +5,7 @@
 
 menuconfig CPLUSPLUS
 	bool "C++ support for the application"
+	select REQUIRES_FULL_LIBC
 	help
 	  This option enables the use of applications built with C++.
```

Comment:
> C++ support requires a toolchain-provided C library with associated libstdc++ library. We don't have a symbol for requiring this, but we can at least exclude using the minimal C library by requiring a full libc.

We actually have `LIB_CPLUSPLUS` for enabling libstdc++ (or any other C++ standard library depending on the toolchain being used). The standard library-less C++ support is actually functional with the minimal libc.

Reply:
Ah, but that configuration doesn't lead to using NEWLIB, it just filters out tests which don't require it through other means. I'll fix this to add a select there and see what you think :-)